test/integration/conftest.py

```diff
@@ -0,0 +1,17 @@
+
+import os
+import pytest
+
+
+@pytest.fixture()
+def git_env():
```

Comment:
I think these environment variables can pass to the subprocess call directly and there is no need to restore the environment state.